source/tests/consistent/model/test_spin_ener.py

```
@@ -384,5 +470,425 @@ def extract_ret(self, ret: Any, backend) -> tuple[np.ndarray, ...]:
+                ret["virial"].ravel(),
+            )
+        elif backend is self.RefBackend.ARRAY_API_STRICT:
+            return (
+                ret["energy"].ravel(),
+                ret["atom_energy"].ravel(),
+                ret["extended_mask_mag"].ravel(),
+                SKIP_FLAG,
+                SKIP_FLAG,
+                SKIP_FLAG,
             )
         raise ValueError(f"Unknown backend: {backend}")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)